src/gui/qgsprovidersublayersdialog.h

```diff
@@ -28,17 +29,30 @@ class QgsProviderSublayerModel;
 class QgsProviderSublayerProxyModel;
 class QgsProviderSublayerTask;
 
-class QgsProviderSublayerDialogModel : public QgsProviderSublayerModel
+/**
+ * \ingroup gui
+ *
+ * \brief A model for representing the sublayers present in a URI for the QgsProviderSublayersDialog.
+ *
+ * \since QGIS 3.30
+ */
+class GUI_EXPORT QgsProviderSublayerDialogModel : public QgsProviderSublayerModel
 {
     Q_OBJECT
 
   public:
 
+    /**
+     * Constructor.
+     */
     QgsProviderSublayerDialogModel( QObject *parent = nullptr );
 
     QVariant data( const QModelIndex &index, int role ) const override;
     Qt::ItemFlags flags( const QModelIndex &index ) const override;
 
+    /**
+     * Sets whether geometry types are resolved.
```

Comment:
If you happen to know, what this refers to, a more verbose version would be appreciated